src/managers/config.py

```diff
@@ -280,14 +280,15 @@ def etc_hosts_entries(self) -> list[str]:
         Returns:
             Multiline string of `/etc/hosts` entries
         """
-        hosts_entries = []
+        current_etc_hosts = self.workload.read(path="/etc/hosts")
+        hosts_entries = set(current_etc_hosts)
         for server in self.state.servers:
             if not all([server.ip, server.hostname, server.fqdn]):
-                return []
+                continue
 
-            hosts_entries.append(f"{server.ip} {server.fqdn} {server.hostname}")
+            hosts_entries.add(f"{server.ip} {server.fqdn} {server.hostname}")
```

Comment:
```suggestion
            for entry in host_entries:
                if server.fqdn in entry or server.hostname in entry:
                    # Clean-up old entry
                    host_entries.remove(entry)
            hosts_entries.add(f"{server.ip} {server.fqdn} {server.hostname}")
```

Reply:
A better fix is provided which covers the IP change cases.